roachtest: improve sqlsmith for an elusive setup error

yuzefovich opened the pull request:
**roachtest: log erroneous stmt in sqlsmith.log**

Previously, if we hit an error when executing a setup query, we would only include it into the error message (so it would only be seen in the test.log file). This commit makes it so that the erroneous setup query is included into sqlsmith.log as well.

Informs: #116160.
Informs: #116307.

**roachtest/sqlsmith: print a setup query as integers**

This commit adjusts the sqlsmith logging so that if it encounters an error during the setup that contains "does not exist" substring, then we will also log the failed stmt as space-separated integers. This change is made in hopes of being able to reproduce the elusive 'pq: column "crdb_internal_idx_expr" does not exist' error we've seen a few times. My hypothesis is that there is some non-visible character that gets lost when stringified.

Epic: None

Release note: None

yuzefovich committed:
roachtest: log erroneous stmt in sqlsmith.log

Previously, if we hit an error when executing a setup query, we would
only include it into the error message (so it would only be seen in the
test.log file). This commit makes it so that the erroneous setup query
is included into sqlsmith.log as well.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/116298)
<!-- Reviewable:end -->

yuzefovich committed:
roachtest/sqlsmith: print a setup query as integers

This commit adjusts the sqlsmith logging so that if it encounters an
error during the setup that contains "does not exist" substring, then we
will also log the failed stmt as space-separated integers. This change
is made in hopes of being able to reproduce the elusive "pq: column
"crdb_internal_idx_expr" does not exist" error we've seen a few times.
My hypothesis is that there is some non-visible character that gets lost
when stringified.

Release note: None

yuzefovich commented:
Friendly ping @mgartner

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/116298#-:-NnAtpqCACMeM1h5iLS8:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

mgartner commented:
Sorry for the delay!

bors r+
